### DISCUSSION TITLE
How to use ImbalancedSampler for edge classfication and node classfication

### DISCUSSION THREAD

**Discussion by lfangyu09** [pyg-team/pytorch_geometric#8901] [General]
Hi,

Thank you so much for providing this wonderful package. I have recently been working on Edge classification and Node classification. It is binary classification but the dataset is quite imbalanced. So I would like to use `ImbalancedSampler` to preprocess the train dataset. But I faced two issues:

(1) Edge classification.
How to use `ImbalancedSampler` to process the train dataset for edge classification? I did these 
```
train_sampler = pyg.loader.ImbalancedSampler(train_dataset)
train_loader = pyg.loader.DataLoader(train_dataset, batch_size=1, shuffle=True, num_workers=2, sampler=train_sampler)
```
But it returns this error:
![image](https://github.com/pyg-team/pytorch_geometric/assets/62622741/38bfe798-e51b-4938-ba76-879805cb7118)
It looks like `ImbalancedSampler` is based on Graph classification. So, how to use `ImbalancedSampler` to process the train dataset for edge?

(2) Node classification.
How to use `ImbalancedSampler` to process the train dataset for node classification? I did these 
```
sampler = ImbalancedSampler(data.y)
loader = NeighborLoader(data, input_nodes=data.train_mask,
                        batch_size=64, num_neighbors=[-1, -1],
                        sampler=sampler, ...)
```
This looks like for each data rather than the whole dataset. So, how to use `ImbalancedSampler` to process the train dataset for nodes?

Thanks a lot.

**Comment by rusty1s** [comment]
1. Currently, `ImbalancedSampler` expects node-level or graph-level labels. You can also use it for edge-level classification, in which case you have to input tensors though, i.e., `ImbalancedSampler(data.edge_label)`.
2. This looks totally fine, right? `NeighborLoader` can only operate on single graphs only anyway.
